gui/wxpython/mapdisp/frame.py

```diff
@@ -974,7 +1000,14 @@ def OnCloseWindow(self, event, askIfSaveWorkspace=True):
             if pgnum_dict is not None:
                 self.CleanUp()
                 if pgnum_dict["layers"] > -1:
-                    self.closingDisplay.emit(pgnum_dict=pgnum_dict)
+                    if self.IsDockable():
+                        self.closingDisplay.emit(
+                            pgnum_dict=pgnum_dict, is_docked=self.IsDocked()
+                        )
+                        if not self.IsDocked():
+                            self.GetParent().Destroy()
```

Comment:
I added the support for workspaces but this error is not related to workspaces (it is just a good way how to notice it) .. it is related to a problem in closing MapPageFrame. The problem is specifically on this row. I started to think that it should be fixed by using signals but not sure how to use them. We would need something like "when closing map display panel automatically raise closing map display frame"...

Reply:
The problem is probably that when undocking the map frame, I remove the page from notebook by RemovePage method but the pointer still exists on this page...

Reply:
The problem is in CallLater, the frame does not exist when Raise is eventually called.

Reply:
Thanks for finding this, I dived too deep into the problem and then was looking for the problem where no problem existed.